git_machete/github.py

```diff
@@ -147,7 +147,9 @@ def __fire_github_api_request(method: str, path: str, token: Optional[str], requ
             return parsed_response_body
     except HTTPError as err:
         if err.code == http.HTTPStatus.UNPROCESSABLE_ENTITY:
-            raise UnprocessableEntityHTTPError(str(err.reason))
+            error_response = json.loads(err.read().decode())
+            error_reason: str = error_response['errors'][0]['message']
```

Comment:
Would it make sense to extract the message in case of 400 status? potentially also 401, 403, 404 (to be checked if github includes anything/anything interesting for these status codes)

Reply:
I don't think so, the errors then are rather self descriptive, see:
- 400 (Bad request) body:  `Bad request`
- 401 (Unauthorised) body: `Bad Credentials`
- 403 (Forbidden) body: `Forbidden for url: https://ap...`
- 404 (Not found) body:  `Not Found`
Also there is no `error` field in response like it is in 422.